helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -19,19 +19,34 @@
  * under the License.
  */
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
+import org.apache.helix.HelixManagerProperties;
+import org.apache.helix.SystemPropertyKeys;
 import org.apache.helix.controller.rebalancer.waged.RebalanceAlgorithm;
 import org.apache.helix.model.ClusterConfig;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 /**
  * The factory class to create an instance of {@link ConstraintBasedAlgorithm}
  */
 public class ConstraintBasedAlgorithmFactory {
+  private static final String PARTITION_MOVEMENT_CONSTRAINT = "PartitionMovementConstraint";
```

Comment:
If preferred, we can put a constraint type in the constraint class. But I don't think that is necessary.
We can just keep it simple for now I guess:
        .put(new PartitionMovementConstraint(), movementRatio * MODEL.get(PartitionMovementConstraint))
        .put(new InstancePartitionsCountConstraint(), evennessRatio * MODEL.get(InstancePartitionsCountConstraint))
        ......

Reply:
We can only extract the string from the property file, and the `PartitionMovementConstraint` variable in `movementRatio * MODEL.get(PartitionMovementConstraint)`, what type is it? This code change is imperfect and I will just let it work at first.

Reply:
How about just use the simple name of the class?

Reply:
Sounds good. Better than hard fixed string IMO